packages/react/menu/src/Menu.tsx

```diff
@@ -565,8 +566,14 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
                     props.onPointerMove,
                     whenMouse((event) => {
                       const target = event.target as HTMLElement;
-                      if (event.currentTarget.contains(target) && event.movementX !== 0) {
-                        pointerDirRef.current = event.movementX > 0 ? 'right' : 'left';
+                      const valueHasChanged = lastPointerXRef.current !== event.clientX;
```

Comment:
Can we perhaps name this `pointerXHasChanged`? It's a bit too generic otherwise.